swap/swap.go

```diff
@@ -532,6 +531,11 @@ func (s *Swap) verifyContract(ctx context.Context, address common.Address) error
 	return contract.ValidateCode(ctx, s.backend, address)
 }
 
+// SetChequebookAddr sets the chequebook address
+func (s *Swap) SetChequebookAddr(chequebookAddr common.Address) {
+	s.owner.Contract = chequebookAddr
```

Comment:
i am assuming the reason for adding this function is either code clarity or extensibility. am i correct in doing so?

Reply:
Only for code clarity indeed. This function is actually redundant as the chequebookAddress is already implicitly set as part of the contract instance at `swap.contract` (from there you can make direct calls to the smart-contract, so it must be set). Nevertheless, the contract address is not publicly exposed in `swap.contract`, so we need some refactor and rewrite.

Reply:
relevant issue: #1728. resolving